cot/src/cli.rs

```diff
@@ -292,7 +292,42 @@ impl CliTask for RunServer {
         };
 
         let bootstrapper = bootstrapper.boot().await?;
-        crate::run(bootstrapper, &addr_port).await
+
+        let result = crate::run(bootstrapper, &addr_port).await;
+        if let Err(error) = &result {
+            if let Some(user_friendly_error) = Self::get_user_friendly_error(error, &addr_port) {
+                eprintln!("{user_friendly_error}");
```

Comment:
If we handled a specific error, shouldn't we exit early here?

Reply:
I think (I might be wrong) that it still makes sense to print out the entire error structure anyways for debugging purposes. Also, this gives a chance to cleanup the stack with RAII as opposed to `std::process::exit` which terminates instantly.